### ISSUE TITLE
Non-static method InsitesConsulting\AzureKeyVault\Vault::secret() cannot be called statically

### ISSUE THREAD

**Issue by carlituxman** [insites-consulting/azure-key-vault#7]
I'm trying this bug using with laravel 8 and have this error

if quit use InsitesConsulting\AzureKeyVault\Vault;
throws this: Undefined array key "access_token"

insites-consulting/azure-key-vault (v0.3.1)

**Comment by shealavington** [comment]
I've been looking into this package myself. I came across the same issue, the cause of the issue is because authentication is failing due to invalid details. 

I found this through normal debugging, however, really, the package should provide an exception if authentication fails, unfortunately it expects a success and doesn't handle any failures.

Some error handling needs adding before this line to fix it: https://github.com/insites-consulting/azure-key-vault/blob/d7ea628c5da3188ebd7e200985a603e77d457c90/src/Vault.php#L43

If I decide to proceed with the package, I might look to PR a fix, else, anyone is welcome to do so.
